### PR TITLE
modifying daq_1Dviewer_LecroyWaverunner6Zi plugin

### PR DESCRIPTION
**Pull request by dplatzer**
In "sequence" mode, the oscilloscope now waits for all the segments before doing acquisition.
It's a minor modification and also the occasion for me to test a pull request.

### CODE REVIEW

**Comment by seb5g** [comment]
Is it fine with David? Make sure he pushed all its modification. Then I'll merge your commit!

File: pymodaq_plugins/daq_viewer_plugins/plugins_1D/daq_1Dviewer_LecroyWaverunner6Zi.py [hunk 1]
```diff
@@ -1,202 +1,223 @@
...
+
+            =============== ======== ===============================================
+            **Parameters**  **Type**  **Description**
+            *Naverage*      int       Number of values to average
+            =============== ======== ===============================================
+        """
+        channel = self.settings.child(('channels')).value()['selected']
+
+        self.controller.WriteString(r"""vbs? 'return=app.acquisition.horizontal.numsegments' """, 1)
+        numSegments = self.controller.ReadString(8)
```
Comment (quantumm) [inline pymodaq_plugins/daq_viewer_plugins/plugins_1D/daq_1Dviewer_LecroyWaverunner6Zi.py]: The numSegments variable (maybe "numberOfSegments" would be more explicit…) seems to be used only in the if sampleMode == "Sequence" condition so I would place those lines in it.

File: pymodaq_plugins/daq_viewer_plugins/plugins_1D/daq_1Dviewer_LecroyWaverunner6Zi.py [hunk 2]
```diff
@@ -1,202 +1,223 @@
...
+
+        self.controller.WriteString(r"""vbs? 'return=app.acquisition.horizontal.samplemode' """, 1)
+        sampleMode = self.controller.ReadString(8)
+
+        # The WaitForOPC method is used to wait for previous commands to be interpreted before continuing
+        # It may be not needed here
+        if not self.controller.WaitForOPC():
+            raise Exception("Wait for OPC error")
+
+        waveform = self.controller.GetScaledWaveformWithTimes(channel[0], 1e8, 0)
```
Comment (quantumm) [inline pymodaq_plugins/daq_viewer_plugins/plugins_1D/daq_1Dviewer_LecroyWaverunner6Zi.py]: I would get the waveform after having waited for acquiredSegments == numSegments no ?

File: pymodaq_plugins/daq_viewer_plugins/plugins_1D/daq_1Dviewer_LecroyWaverunner6Zi.py [hunk 3]
```diff
@@ -1,202 +1,223 @@
...
+                self.controller = active_dso
+                usb_address = "USBTMC:" + self.settings.child(('VISA_ressources')).value()
+                self.controller.MakeConnection(usb_address)
+                # set the timeout of the scope to 10 seconds
+                # may be not needed
+                self.controller.setTimeout(10)
+
+            self.status.initialized = True
+            self.status.controller = self.controller
+            # self.controller.WriteString(r"""vbs 'app.acquisition.triggermode = "stopped" ' """, 1)
```
Comment (quantumm) [inline pymodaq_plugins/daq_viewer_plugins/plugins_1D/daq_1Dviewer_LecroyWaverunner6Zi.py]: This has been added for tests I suppose

Comment (seb5g) [reply pymodaq_plugins/daq_viewer_plugins/plugins_1D/daq_1Dviewer_LecroyWaverunner6Zi.py]: what should I do then, merge it into your plugin?

Comment (quantumm) [reply pymodaq_plugins/daq_viewer_plugins/plugins_1D/daq_1Dviewer_LecroyWaverunner6Zi.py]: wait for the author of the commit to make the corrections before merging

**Pull request merged**